### PR TITLE
Add useragent to handlers

### PR DESCRIPTION
**Pull request by rmweir**
Some handlers have been configured to use clients with useragents that are named to reference the handlers that use them.

## Issue: 

https://github.com/rancher/rancher/issues/38775

## Problem

We would like to use the new useragent functionality in a small way that only mainly only affects team1 before rolling it out to more parts of rancher.
 
## Solution

Add useragent to some select handlers.

### CODE REVIEW

File: pkg/controllers/management/auth/register.go [hunk 1]
```diff
@@ -56,14 +56,14 @@ func RegisterIndexers(scaledContext *config.ScaledContext) error {
...
 
 func RegisterEarly(ctx context.Context, management *config.ManagementContext, clusterManager *clustermanager.Manager) {
-	prtb, crtb := newRTBLifecycles(management)
-	gr := newGlobalRoleLifecycle(management)
-	grb := newGlobalRoleBindingLifecycle(management, clusterManager)
+	prtb, crtb := newRTBLifecycles(management.WithAgent("mgmt-auth-crtb-prtb-controller"))
+	gr := newGlobalRoleLifecycle(management.WithAgent(grController))
+	grb := newGlobalRoleBindingLifecycle(management.WithAgent(grbController), clusterManager)
 	p, c := newPandCLifecycles(management)
 	u := newUserLifecycle(management, clusterManager)
```
Comment (cmurphy) [inline pkg/controllers/management/auth/register.go:63]: I know the plan isn't to roll this out everywhere but I'm not clear why some of these were chosen and not others, for instance why do we want this for user attributes but not yet for users?

File: pkg/controllers/managementuser/rbac/handler_base.go [hunk 2]
```diff
@@ -116,11 +118,11 @@ func Register(ctx context.Context, workload *config.UserContext) {
...
-		ProjectLister:       workload.Management.Management.Projects(workload.ClusterName).Controller().Lister(),
+		ProjectLister:       management.Management.Projects(workload.ClusterName).Controller().Lister(),
 	}
 
 	workload.Core.Namespaces("").AddLifecycle(ctx, "namespace-auth", newNamespaceLifecycle(r, sync))
-	workload.Management.Management.RoleTemplates("").AddHandler(ctx, "cluster-roletemplate-sync", newRTLifecycle(r))
+	management.Management.RoleTemplates("").AddHandler(ctx, "cluster-roletemplate-sync", newRTLifecycle(r))
 }
 
 type manager struct {
```
Comment (cbron) [inline pkg/controllers/managementuser/rbac/handler_base.go:128]: Are we only doing 2.6 and not 2.7 for the test run ?

Comment (rmweir) [reply pkg/controllers/managementuser/rbac/handler_base.go:128]: I'm planning on doing both and am partway through forward porting to 2.7. Just waiting on this.

File: pkg/wrangler/context.go [hunk 3]
```diff
@@ -456,6 +456,9 @@ func NewContext(ctx context.Context, clientConfig clientcmd.ClientConfig, restCo
 		core:         core,
 		api:          api,
 		crd:          crd,
+		capi:         capi,
+		rke:          rke,
+		rbac:         rbac,
```
Comment (cbron) [inline pkg/wrangler/context.go:461]: what's the point of adding these ?

Comment (rmweir) [reply pkg/wrangler/context.go:461]: It's a fix, I missed them [here](https://github.com/rancher/rancher/pull/39876/files#diff-a386241816deb9f14c0dbac3c18f0642a47992060fe0f582782dd8be506d2c31R151-R164).

**Review by cbron — APPROVED** [review]

**Review by cmurphy — APPROVED** [review]

**Pull request merged**